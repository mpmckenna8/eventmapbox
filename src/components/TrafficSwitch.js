import PropTypes from 'prop-types';
import React, {Component} from 'react';
import {connect} from 'react-redux';
import {triggerMapUpdate, setStateValues} from '../actions/index';

class TrafficSwitch extends Component {
  render() {
    return (
      <label className='switch-container align-middle px3 bg-lighten75'>
        <input type='checkbox'
          onChange={(e) => this.toggle(e, this.trafficInStyle())}
          checked={this.trafficInStyle()}
        />
        <div className='switch switch--blue'></div>
        <svg className='icon icon-switch color-blue'><use xlinkHref='#icon-street'/></svg>
      </label>
    );
  }

  toggle(e, trafficInStyle) {
    var checked = e.target.checked;
    if (checked && !trafficInStyle) {
      this.props.setStateValues({
        mapStyle: this.props.mapStyle + '-traffic',
        needMapRestyle: true
      });

      this.props.triggerMapUpdate();
    } else if (!checked && trafficInStyle) {
      this.props.setStateValues({
        mapStyle: this.props.mapStyle.split('-')[0],
        needMapRestyle: true
      });

      this.props.triggerMapUpdate();
    }
  }

  trafficInStyle() {
    return this.props.mapStyle.indexOf('traffic') > -1;
  }
}

TrafficSwitch.propTypes = {
  setStateValues: PropTypes.func,
  mapStyle: PropTypes.string,
  triggerMapUpdate: PropTypes.func,
};

const mapStateToProps = (state) => {
  return {
    mapStyle: state.app.mapStyle,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    setStateValues: (obj) => dispatch(setStateValues(obj)),
    triggerMapUpdate: (v) => dispatch(triggerMapUpdate(v))
  };
};

export {TrafficSwitch};
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(TrafficSwitch);
