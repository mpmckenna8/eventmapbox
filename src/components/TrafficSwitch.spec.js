import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import {TrafficSwitch} from './TrafficSwitch';

it('renders correctly', () => {

  const component = renderer.create(
    <TrafficSwitch
      triggerMapUpdate={() => {}}
      mapStyle='streets'
      setStateValues={() => {}}
    />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('switches style on change event', () => {
  const setStateValues = jest.fn();

  const trafficSwitch = shallow(
    <TrafficSwitch
      triggerMapUpdate={() => {}}
      mapStyle='streets'
      setStateValues={setStateValues}
    />
  );

  trafficSwitch.find('input').simulate('change', {target: {checked: true}});

  expect(setStateValues).toBeCalled();
  expect(setStateValues).toBeCalledWith({mapStyle: 'streets-traffic', needMapRestyle: true});
});
