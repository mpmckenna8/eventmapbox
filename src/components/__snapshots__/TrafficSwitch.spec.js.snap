exports[`test renders correctly 1`] = `
<label
  className="switch-container align-middle px3 bg-lighten75">
  <input
    checked={false}
    onChange={[Function]}
    type="checkbox" />
  <div
    className="switch switch--blue" />
  <svg
    className="icon icon-switch color-blue">
    <use
      xlinkHref="#icon-street" />
  </svg>
</label>
`;
